From 092a953cedd9afb5beaf2228b758e26cfb725582 Mon Sep 17 00:00:00 2001
From: Thomas Tai <thomas.tai@windriver.com>
Date: Wed, 2 Mar 2011 10:08:40 -0500
Subject: [PATCH 1/4] cert/Kconfig: add configure options to enable testing frame start and end interrupts

The certifiable hypervisor provides optional start and end frame interrupts.
The start frame interrupt is asserted when hypervisor schedule the VB
to run. The end of frame interrupt is asserted when the VB reach the
end of the schedule period.

Signed-off-by: Thomas Tai <thomas.tai@windriver.com>
---
 drivers/wrhv/Kconfig  |    8 ++++++++
 drivers/wrhv/Makefile |    1 +
 2 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/wrhv/Kconfig b/drivers/wrhv/Kconfig
index bf93abd..5d2169c 100644
--- a/drivers/wrhv/Kconfig
+++ b/drivers/wrhv/Kconfig
@@ -8,6 +8,14 @@ menuconfig WRHV_VIRTIO
 
 if WRHV_VIRTIO
 
+config WRHV_FRAME_TRANSITION_INTERRUPT_TEST
+	bool "Frame transition interrupt test"
+	depends on WRHV_CERT
+	default n
+	---help---
+	  Enable testing the start frame and end frame transition interrupt
+	  for Certifiable Hypervisor.
+
 config WRHV_VIRTIO_SERIAL
 	bool "Virtual serial device support for WindRiver Hypervisor Guest OS"
 	depends on !WRHV_DUART && !SERIAL_8250
diff --git a/drivers/wrhv/Makefile b/drivers/wrhv/Makefile
index 810dc74..bae7d0f 100644
--- a/drivers/wrhv/Makefile
+++ b/drivers/wrhv/Makefile
@@ -1,5 +1,6 @@
 obj-$(CONFIG_WRHV_VIRTIO)	+= wrhv_devices.o
 obj-$(CONFIG_WRHV_DUART)	+= wrhv_duart.o
 obj-$(CONFIG_WRHV_VIRTIO_SERIAL) += wrhv_serial_driver.o
+obj-$(CONFIG_WRHV_FRAME_TRANSITION_INTERRUPT_TEST) += wrhv_frame_irq.o
 
 wrhv_serial_driver-objs	:= wrhv_serial.o wrhv_duart.o
-- 
1.6.5.2

